src/gui/layertree/qgslayertreeview.cpp

```diff
@@ -507,6 +507,23 @@ void QgsLayerTreeView::mouseReleaseEvent( QMouseEvent *event )
 
 void QgsLayerTreeView::keyPressEvent( QKeyEvent *event )
 {
+  if ( event->key() == Qt::Key_Space )
+  {
+    const auto constSelectedNodes = selectedNodes();
+
+    if ( ! constSelectedNodes.isEmpty() )
+    {
+      bool isFirstNodeChecked = constSelectedNodes[0]->itemVisibilityChecked();
+      for ( QgsLayerTreeNode *node : constSelectedNodes )
+      {
+        node->setItemVisibilityChecked( ! isFirstNodeChecked );
+      }
+
+      // if we call the original keyPress handler, the current item will be checked to the original state yet again
+      return;
```

Comment:
If space is removed from the global shortcut this should not be necessary

Reply:
Space is not a global shortcut. Unfortunately it is still needed.